src/plugins/intel_npu/src/plugin/npuw/just_sync_infer_request.cpp

```diff
@@ -168,6 +169,37 @@ ov::npuw::JustInferRequest::JustInferRequest(const std::shared_ptr<ov::npuw::Com
         m_subrequests_gio.at(sub_idx).global_results[i] = out_idx;
     }
     // }}}
+
+    // Sort out how to handle weights bank and unpack
+    for (size_t i = 0; i < m_num_submodels; i++) {
+        LOG_VERB("Trying to preemptively set tensors for Subgraph[" << i << "]...");
+        auto& comp_model_desc = m_npuw_model->m_compiled_submodels[i];
+
+        if (!comp_model_desc.compiled_model && !comp_model_desc.replaced_by) {
+            continue;
+        }
+
+        const auto real_idx = comp_model_desc.replaced_by.value();
+        auto& func_desc = m_npuw_model->m_compiled_submodels[real_idx];
+
+        auto& request = m_subrequests[real_idx];
+
+        for (std::size_t cidx = 0u; cidx < comp_model_desc.closure.size(); cidx++) {
+            const auto& closure = comp_model_desc.closure[cidx];
+
+            const auto closure_param_id = comp_model_desc.param_base + cidx;
+            const auto& iport = func_desc.compiled_model->inputs()[closure_param_id];
+            const auto& clparam = request->get_tensor(iport);
+
+            // No update required means no folding
+            // In this case we can set tensor only once
+            if (!comp_model_desc.update_required[cidx]) {
+                request->set_tensor(iport, ov::get_tensor_impl(closure));
+                LOG_VERB("Tensor set");
```

Comment:
Don't think that such message alone is useful. You don't even write which tensor is set (index, type, etc)

Reply:
Removed